File: programs/advanced-topics_event-listeners-1.cpp
class Sprite : public Actor, public KeyListener {
private:
  int xpos;
  int ypos;
  int dx;
  int dy;
  Image image;

public:
  Sprite(const string& path, int xpos, int ypos): xpos(xpos), ypos(ypos) {
    try {
      this->image = ImageIO::read(new File(path.c_str()));
    } catch (const IOException& exc) {
      exc.printStackTrace();
    }
  }
};

File: programs/advanced-topics_event-listeners-1.java
public class Sprite implements Actor, KeyListener {
    private int xpos;
    private int ypos;
    private int dx;
    private int dy;
    private Image image;

    public Sprite(String path, int xpos, int ypos) {
        this.xpos = xpos;
        this.ypos = ypos;
        try {
            this.image = ImageIO.read(new File(path));
        } catch (IOException exc) {
            exc.printStackTrace();
        }
    }
}